### PR TITLE
Feature/dashboard

### PR DESCRIPTION
**Pull request by dmapel**
update folder structure to follow dev
added new dashboard features

### CODE REVIEW

File: src/app/routes/dashboard/dashboard.component.html [hunk 1]
```diff
@@ -64,36 +68,82 @@ <h1>
...
+          <!-- Position Column -->
+          <ng-container matColumnDef="idno">
+            <th mat-header-cell *matHeaderCellDef> ID </th>
+            <td mat-cell *matCellDef="let sub"> {{sub.id}} </td>
+          </ng-container>
+
+          <ng-container matColumnDef="control">
+            <th mat-header-cell *matHeaderCellDef> Controls </th>
+            <td mat-cell *matCellDef="let sub"><button class='btn btn-outline-dark' (click)='growl(sub.id)'>
+                <mat-icon>done</mat-icon>
```
Comment (danielsidhu) [inline src/app/routes/dashboard/dashboard.component.html:101]: line 123 in the dashboard component html file needs to be shifted over

File: src/app/routes/dashboard/dashboard.component.html [hunk 2]
```diff
@@ -64,36 +68,82 @@ <h1>
...
+          <!-- Name Column -->
+          <ng-container matColumnDef="sub">
+            <th mat-header-cell *matHeaderCellDef> Name </th>
+            <td mat-cell *matCellDef="let sub"> {{sub.name}} </td>
+          </ng-container>
+
+          <!-- Weight Column -->
+
+
+          <ng-container matColumnDef="time">
```
Comment (danielsidhu) [inline src/app/routes/dashboard/dashboard.component.html]: dashboard.html file one line 135 to 138 has extra white space

**Review by danielsidhu — DISMISSED** [review]
some excess white spaces needs to be cleaned up down to one line spaced around each method/block

File: src/app/shared-components/navbar/navbar.component.css [hunk 3]
```diff
@@ -1,45 +1,9 @@
...
-    color: salmon !important;
-    border-bottom: 2px solid salmon !important;
-}
-
 mat-toolbar {
     background-color: rgb(156, 156, 156, 0) !important;
-}
\ No newline at end of file
+}
+/* .mat-dialog-container {
```
Comment (humphreyminottjr) [inline src/app/shared-components/navbar/navbar.component.css]: no commented out code navbar.component.css line 4 to line 9

**Review by humphreyminottjr — DISMISSED** [review]
no commented out code navbar.component.css line 4 to line 9

**Review by YamasakiRhys — DISMISSED** [review]
For NgDocs, please add the authors of the code with the batch and trainer

File: src/app/routes/dashboard/dashboard.component.html [hunk 4]
```diff
@@ -1,171 +1,180 @@
...
+                  <span class="badge badge-success">{{this.percentCompletion | percent}} Completed</span>
+                </p>
+
             </mat-list-item>
-          </mat-list>
-          <mat-progress-bar mode="determinate" [value]="this.percentCompletion * 100" ></mat-progress-bar>
-        </mat-card-content>
+            <mat-divider></mat-divider>
+        
+          </mat-list> -->
```
Comment (MarcinSalamon) [inline src/app/routes/dashboard/dashboard.component.html]: @shannonjeg please delete commented out lines before merging

File: src/app/routes/dashboard/dashboard.component.html [hunk 5]
```diff
@@ -1,171 +1,180 @@
...
+
+        <!-- </mat-card-content> -->
+
       </mat-card>
-    </div>
-  </div>
+
+
+
```
Comment (MarcinSalamon) [inline src/app/routes/dashboard/dashboard.component.html]: too much white space

File: src/app/routes/dashboard/dashboard.component.ts [hunk 6]
```diff
@@ -30,6 +58,12 @@ export class DashboardComponent implements OnInit {
   editing = false;
   firstName: string;
   lastName: string;
+  isOpen = 0;
```
Comment (MarcinSalamon) [inline src/app/routes/dashboard/dashboard.component.ts]: should be a boolean

File: src/app/shared-components/user-info/user-info.component.ts [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+    private router: Router,
+    private batchService: BatchService,
+    private userService: UserService
+  ) { }
+
+  /**
+  * This method runs when the component is initialized. It will get the user
+  * data from the session storage and display both the user's personal info,
+  * and info about the batch they are associated with.
+  */
```
Comment (YamasakiRhys) [inline src/app/shared-components/user-info/user-info.component.ts:41]: Add @author following the correct format:  name | batch | trainer name

File: src/app/routes/dashboard/dashboard.component.ts [hunk 8]
```diff
@@ -30,6 +58,12 @@ export class DashboardComponent implements OnInit {
...
+  visibilityIcon = [{num: 0, icon: 'visibility_off' },
+                    {num: 1, icon: 'visibility'}];
+  DashTitle = 'Today';
+
+
```
Comment (MarcinSalamon) [inline src/app/routes/dashboard/dashboard.component.ts:67]: delete the white space

File: src/app/routes/dashboard/dashboard.component.ts [hunk 9]
```diff
@@ -82,12 +117,22 @@ export class DashboardComponent implements OnInit {
     }
   }
 
+
+  growl(x) {
+    console.log(x);
+       }
+
```
Comment (MarcinSalamon) [inline src/app/routes/dashboard/dashboard.component.ts:124]: please delete before dev

File: src/app/shared-components/user-info/user-info.component.ts [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+  ngOnInit() {
+    this.user = JSON.parse(sessionStorage.getItem('user'));
+  }
+
+  /**
+   * This method toggles editing mode on the personal information
+   * card.
+   */
```
Comment (YamasakiRhys) [inline src/app/shared-components/user-info/user-info.component.ts:49]: @author needed

File: src/app/routes/dashboard/dashboard.component.ts [hunk 11]
```diff
@@ -82,12 +117,22 @@ export class DashboardComponent implements OnInit {
...
   /**
    * This method is used t sort throught a list of batches. Batches
    * are sorted by their start dates.
    * @param a A batch
    * @param b Another batch
    */
+
+
+
```
Comment (MarcinSalamon) [inline src/app/routes/dashboard/dashboard.component.ts:135]: delete white spaces

File: src/app/shared-components/user-info/user-info.component.spec.ts [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+  beforeEach(() => {
+    fixture = TestBed.createComponent(UserInfoComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+});
```
Comment (MarcinSalamon) [inline src/app/shared-components/user-info/user-info.component.spec.ts:25]: please create comments for the code

File: src/app/shared-components/user-info/user-info.component.ts [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+  saveChanges() {
+    this.userService.updateInfo(this.user).subscribe(
+      result => {
+        sessionStorage.setItem('user', JSON.stringify(this.user));
+      }
+    );
+    this.editing = false;
+  }
+}
```
Comment (MarcinSalamon) [inline src/app/shared-components/user-info/user-info.component.ts:77]: put @ author on your methods

**Review by MarcinSalamon — CHANGES_REQUESTED** [review]

File: src/app/shared-components/user-info/user-info.component.spec.ts (hunk 12 above)
Comment (dmapel) [reply src/app/shared-components/user-info/user-info.component.spec.ts:25]: send to Chris to address testing

File: src/app/shared-components/user-info/user-info.component.ts [hunk 14]
```diff
@@ -10,7 +10,7 @@ import { UserService } from '../../services/user.service';
  * user is directed to after they login. It displays personal
  * information and batch information to the user.
  *
- * @author Bradley Walker | Khaleel Williams | 1806-Jun18-USF-Java | Wezley Singleton
```
Comment (MarcinSalamon) [inline src/app/shared-components/user-info/user-info.component.ts]: don't delete previous authors

File: src/app/routes/dashboard/dashboard.component.html [hunk 15]
```diff
@@ -143,38 +119,12 @@ <h1>
...
+    <mat-sidenav #tsidenav mode="side" position="end">
+      <app-topics></app-topics>
+    </mat-sidenav>
 
-
-
-    </mat-sidenav-content>
   </mat-sidenav-container>
 
 </div>
```
Comment (MarcinSalamon) [inline src/app/routes/dashboard/dashboard.component.html]: add end line

File: src/app/routes/topics/topics.component.spec.ts [hunk 16]
```diff
@@ -0,0 +1,25 @@
...
+  beforeEach(() => {
+    fixture = TestBed.createComponent(TopicsComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+});
```
Comment (MarcinSalamon) [inline src/app/routes/topics/topics.component.spec.ts:25]: create documentation

**Review by MarcinSalamon — DISMISSED** [review]

**Review by humphreyminottjr — COMMENTED** [review]
Good!

**Review by danielsidhu — APPROVED** [review]
changes have been resolved and code looks good

**Pull request merged**